File: src/jvm/debug_var.c
/*
 * J2ME Emulator - Debug runtime variables and logging mutex
 */

#include <stdio.h>

#ifdef _WIN32
#define WIN32_LEAN_AND_MEAN
#include <windows.h>
static CRITICAL_SECTION g_log_cs;
static int g_log_initialized = 0;
#else
#include <pthread.h>
static pthread_mutex_t g_log_mutex = PTHREAD_MUTEX_INITIALIZER;
#endif

int g_j2me_runtime_debug = 0;

void log_mutex_init(void) {
#ifdef _WIN32
    if (!g_log_initialized) {
        InitializeCriticalSection(&g_log_cs);
        g_log_initialized = 1;
    }
#endif
}

void log_lock(void) {
#ifdef _WIN32
    if (!g_log_initialized) {
        InitializeCriticalSection(&g_log_cs);
        g_log_initialized = 1;
    }
    EnterCriticalSection(&g_log_cs);
#else
    pthread_mutex_lock(&g_log_mutex);
#endif
}

void log_unlock(void) {
#ifdef _WIN32
    LeaveCriticalSection(&g_log_cs);
#else
    pthread_mutex_unlock(&g_log_mutex);
#endif
}

File: js/jvm/debug_var.mjs
/*
 * J2ME Emulator - Debug runtime variables and logging mutex
 *
 * Migrated from src/jvm/debug_var.c
 *
 * Translation notes:
 *   - `int g_j2me_runtime_debug` becomes a boxed object { value: 0 } so that
 *     importers can hold a reference and observe mutations (JS has no shared
 *     mutable primitive across module boundaries).
 *   - log_lock / log_unlock / log_mutex_init are no-ops: Node.js is
 *     single-threaded and has no need for a logging mutex.
 *   - The Windows CRITICAL_SECTION and POSIX pthread_mutex_t are both omitted.
 */

/**
 * Runtime debug flag.
 *
 * Mirrors `int g_j2me_runtime_debug` from debug_var.c.
 * Use `.value` to read or write:
 *
 *   import { g_j2me_runtime_debug } from './debug_var.mjs';
 *   if (g_j2me_runtime_debug.value) { ... }
 *   g_j2me_runtime_debug.value = 1;  // enable
 */
export const g_j2me_runtime_debug = { value: 0 };

/**
 * Initialize the logging mutex.
 * No-op in JavaScript (single-threaded runtime, no mutex needed).
 * Exported to preserve the symbol declared in debug.h / debug_macros.h.
 */
export function log_mutex_init() {
    // no-op
}

/**
 * Acquire the logging mutex.
 * No-op in JavaScript.
 */
export function log_lock() {
    // no-op
}

/**
 * Release the logging mutex.
 * No-op in JavaScript.
 */
export function log_unlock() {
    // no-op
}
